src/PiWeb.Import.Sdk/Modules/ImportFormat/IFileSource.cs

```diff
@@ -0,0 +1,76 @@
+﻿#region copyright
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+/* Carl Zeiss Industrielle Messtechnik GmbH        */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2024                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+namespace Zeiss.PiWeb.Import.Sdk.Modules.ImportFormat;
+
+#region usings
+
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+
+#endregion
+
+/// <summary>
+/// Represents a file.
+/// </summary>
+public interface IFileSource
+{
+    #region properties
+
+    /// <summary>
+    /// Gets the complete file name (including the extension) of this <see cref="IFileSource"/>.
+    /// </summary>
+    public string Name { get; }
+
+    /// <summary>
+    /// Gets the file name (without extension) of this <see cref="IFileSource"/> or <c>null</c> if the file has no name.
+    /// </summary>
+    public string? BaseName { get; }
```

Comment:
In what cases would we return null as base name? I think this will always have a valid value as long name has a valid value.